include/deepworks/optimizer.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <deepworks/tensor.hpp>
+#include <deepworks/parameter.hpp>
+
+namespace deepworks {
+namespace optimizer {
+
+class SGD {
+public:
+
+    explicit SGD(Parameters& params, float lr);
+
+    void step();
+
+    float get_learning_rate();
+
+    void set_learning_rate(float lr);
+
+private:
+    float learning_rate;
```

Comment:
to keep consistency I would name it `m_lr` or `m_learning_rate` and `m_parameters` or `m_params`